src/main/java/Duke.java

```diff
@@ -1,10 +1,233 @@
+        loadTasksFromFile();
+
+        // Greet the user and initialise Jonas Chatbot
+        String name = "Jonas";
+        Scanner scanner = new Scanner(System.in);
+
+        System.out.println("____________________________________________________________");
+        System.out.println("Hello! I'm " + name + ".");
+        System.out.println("What can I do for you?");
+        System.out.println("____________________________________________________________");
+
+        // Read user input
+        while (true) {
+            String userInput = scanner.nextLine();
+
+            if (userInput.equalsIgnoreCase("bye")) {
+                System.out.println("____________________________________________________________");
+                System.out.println("Kamxia. Hope to see you again soon!");
+                System.out.println("____________________________________________________________");
+                saveTasksToFile();
+                break;
+            } else {
+                try {
+                    processInput(userInput);
+                } catch (DukeException e) {
+                    System.out.println("____________________________________________________________");
+                    System.out.println("OOPS!!! " + e.getMessage());
+                    System.out.println("____________________________________________________________");
+                }
+            }
+        }
+
+        scanner.close();
+    }
+
+    // Process user input
+    private static void processInput(String userInput) throws DukeException {
+        if (userInput.startsWith("todo ")) {
+            addTodoTask(userInput.substring(5));
+        } else if (userInput.startsWith("deadline ")) {
+            String[] split = userInput.substring(9).split(" /by ");
+            if (split.length != 2) {
+                throw new DukeException("The description of a deadline task should be followed by '/by <deadline>'.");
+            }
+            addDeadlineTask(split[0], split[1]);
+        } else if (userInput.startsWith("event ")) {
+            String[] split = userInput.substring(6).split(" /from | /to ");
+            if (split.length != 3) {
+                throw new DukeException("The description of an event task should be followed by '/from <start> /to <end>'.");
+            }
+            addEventTask(split[0], split[1], split[2]);
+        } else if (userInput.equalsIgnoreCase("list")) {
+            displayTasks();
+        } else if (userInput.startsWith("mark ")) {
+            markTask(userInput);
+        } else if (userInput.startsWith("unmark ")) {
+            unmarkTask(userInput);
+        } else if (userInput.startsWith("delete ")) {
+            deleteTask(userInput);
+        } else {
+            throw new DukeException("I'm sorry, but I don't know what that means :(");
+        }
+    }
+
+    // Add a todo task to the list
+    public static void addTodoTask(String description) throws DukeException {
```

Comment:
Can create a command abstract class similar to how it is done in personsbook for executing the different commands (add, delete, etc)